### PR TITLE
Updates for 0.15

### PR DESCRIPTION
**Pull request by sonnym**
This makes some headway, but still does not compile.  I am now getting this issue:

```
Last 10 lines of the build log ( /home/sonny/.elmenv/elm-platform/installers/Elm-Platform/master/logs/elm-reactor-0.3.log ):
Building elm-reactor-0.3...
Preprocessing executable 'elm-reactor' for elm-reactor-0.3...
Failed to build SideBar.elm

your elm-package.json file says you depend on package evancz/elm-markdown,
but it looks like it is not properly installed. Try running 'elm-package install'.


Custom build step: creating and collecting all static resources
cabal: Error: some packages failed to install:
elm-reactor-0.3 failed during the building phase. The exception was:
ExitFailure 1
```

My initial guess is that there is no updated version of `elm-markdown` for 0.15 yet.  Would this make sense?

### CODE REVIEW

**Comment by evancz** [comment]
Thanks for working on this! :D

Hmm, I wonder if the version constraints are stale for the whole elm-package.json file? I think they should match [these constraints](https://github.com/elm-lang/elm-lang.org/blob/master/elm-package.json#L12-L15). Does that help?

**Comment by sonnym** [comment]
No change in terms of the error generated.  I also tried again after deleting `~/.elm/packages` to no avail.  I have included those updated constraints, though, since they are probably necessary anyway.

**Comment by evancz** [comment]
[elm-markdown is updated](https://github.com/evancz/elm-markdown/blob/master/elm-package.json#L17) and I am using it successfully for the package website now. Have you tried getting rid of elm-stuff/ in case something bad got in there?

Also, when you run the command `elm-package` you should see this:

```
Elm Package Manager 0.5 (Elm Platform 0.15)
```

**Comment by sonnym** [comment]
Yep, tried that as well.  I was actually using the wrong `elm-package` initially, which made me hopeful, but using the correct one (and removing the two directories) still fails in the same way.

**Comment by evancz** [comment]
I just fixed the elm-package.json file and pushed it, so it should work now. I didn't realize this was a PR until just now, but I think the problem was the constraint for core which used 2.0.0 for the upper and lower bound.

**Comment by evancz** [comment]
The SideBar.elm file still has problems though! :)

**Comment by evancz** [comment]
I was thinking maybe we found a bug in elm-package, but here's the problem: `2.0.0 <= v <= 2.0.0` would have been fine, but in `2.0.0 <= v < 2.0.0` you demand that it is both greater than or equal to `2.0.0` and that it is *less* than `2.0.0`.

**Comment by sonnym** [comment]
Cool, that fixed it!  I have actually made that mistake before, and it is quite confusing how it silently fails.  Would you like me to open an issue on elm-package for that?  I am also currently looking at updating Sidebar.elm.

**Comment by evancz** [comment]
I think it'd be good to have an issue. I guess the problem is "error messages are bad when constraint solving fails" though I'm not sure how to address that in a nice way at the moment :)

**Comment by sonnym** [comment]
Cool, I mentioned this particular case on an existing issue that seems to be due to the same underlying problem.  I've updated the import syntax, but now need to go through and update the usage of `Signal.Message`, which touches `Native.Slider`, so that is going to be a bit more involved.

**Comment by sonnym** [comment]
I keep running into errors like the following, which make no sense to me:

```
Error in ./frontend/SideBar/Watches.elm:

Error on line 14, column 14 to 22:
Ambiguous usage of variable 'List.map'.
    Disambiguate between: List.map, List.map
```

https://github.com/elm-lang/elm-reactor/blob/master/frontend/SideBar/Watches.elm#L15

I am able to work around the issue by un-qualifying the function call, but that seems less than optimal.

**Comment by evancz** [comment]
[The default imports](https://github.com/elm-lang/core#default-imports) include `List` now, so you can say `List.map` anywhere. The correct fix is to remove the redundant import line. Unqualifiying things is definitely not the answer.

In the latest version of elm-compiler, it handles these things okay, but I'd recommend fixing it how things are now before upgrading. That way you'll remove any redundant imports.

**Comment by sonnym** [comment]
Ah, that makes sense!  Good to hear that `elm-compiler` will now handle redundant imports, but I'll go back through and clean things up such that they don't exist in the first place.

**Comment by sonnym** [comment]
I've made some more headway, but now I'm running into an error that has me stumped.

```
Building elm-reactor-0.3...
Preprocessing executable 'elm-reactor' for elm-reactor-0.3...
Failed to build SideBar.elm


Error in frontend/SideBar.elm:

Error on line 34, column 14 to 38:
Could not find variable 'Controls.permitSwapMailbox.signal'.

Looks like the prefix 'Controls.permitSwapMailbox' is not in scope. Is it spelled correctly?
Is it imported correctly?

You can read about how imports work at the following address:
<http://elm-lang.org/learn/Syntax.elm#modules>


Custom build step: creating and collecting all static resources
```

But the module is totally imported: https://github.com/sonnym/elm-reactor/blob/a26b1d8e3120990816358900004cca1d442cd7a7/frontend/SideBar.elm#L8

And the function totally exists:  https://github.com/sonnym/elm-reactor/blob/a26b1d8e3120990816358900004cca1d442cd7a7/frontend/SideBar/Controls.elm#L133

**Comment by evancz** [comment]
Interesting, this looks like a compiler bug. It is treating the whole thing as a variable, instead of treating `Controls.permitSwapMailbox` as a variable and `.signal` as a record access. I wonder if this exists in 0.14?

In any case, try putting parens like this to verify the bug: `(Controls.permitSwapMailbox).signal`

**Comment by sonnym** [comment]
Adding parentheses doesn't workaround the issues. I get the same error.

**Comment by sonnym** [comment]
@evancz 

I figured out a nice and simple work around by simply making new functions to refer to the ones from the `Controls` module.  After this most recent commit, everything compiles and runs fine!

File: frontend/SideBar/Watches.elm [hunk 1]
```diff
@@ -1,9 +1,7 @@
 module SideBar.Watches where
 
-import Html
-import Html (..)
-import Html.Attributes (..)
-import List
+import Html as Html exposing (..)
```
Comment (evancz) [inline frontend/SideBar/Watches.elm]: This line is redundant, writing `import Html exposing (..)` does the same thing

File: frontend/SideBar/Controls.elm [hunk 2]
```diff
@@ -190,13 +188,13 @@ sliderMinMaxText : Int -> Model.Model -> Element
 sliderMinMaxText w state =
     let sliderStartText =
             textStyle "0"
-                |> Text.leftAligned
+                |> GE.leftAligned
```
Comment (evancz) [inline frontend/SideBar/Controls.elm]: Will this collide with something if it is not prefixed?

Comment (sonnym) [inline frontend/SideBar/Controls.elm]: Nope! https://github.com/sonnym/elm-reactor/commit/364899717590b1be2293091cde629b8f50d939da

File: frontend/SideBar/Watches.elm (hunk 1 above)
Comment (sonnym) [inline frontend/SideBar/Watches.elm]: Ah, I wasn't entirely clear on the new import syntax.  Very nice!  Fixed now.

**Comment by evancz** [comment]
Awesome, thank you!

When you tried the parens solution, did you try it in one place or all of them? What happens if you add it in one place now that things are working now? It feels like something odd is happening there.

**Comment by sonnym** [comment]
No worries!  I switched everything over to using additional parentheses, and it still failed on the first instance of the problem.  During implementing the workaround, I missed a few places at one point, and the error was repeated for those lines, so it seems like any place you try to do `Module.function.recordMember` is somehow broken.

**Pull request merged**